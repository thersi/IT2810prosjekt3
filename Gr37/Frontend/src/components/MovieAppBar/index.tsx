import AppBar from "@mui/material/AppBar";
import Box from "@mui/material/Box";
import Toolbar from "@mui/material/Toolbar";
import IconButton from "@mui/material/IconButton";
import Typography from "@mui/material/Typography";
import MenuIcon from "@mui/icons-material/Menu";
import SearchBar from "../SearchBar";
import MovieFilter from "../MovieFilter";
import { createTheme } from "@mui/system";

export default function MovieAppBar() {
  const darkTheme = createTheme({
    palette: {
      mode: "dark",
    },
  });
  return (
    <Box sx={{ flexGrow: 1 }}>
      <AppBar position="static">
        <Toolbar>
          <Typography
            variant="h6"
            noWrap
            component="div"
            sx={{ flexGrow: 1, display: { xs: "none", sm: "block" } }}
          >
            Movies
          </Typography>
          <MovieFilter />
          <SearchBar />
        </Toolbar>
      </AppBar>
    </Box>
  );
}
