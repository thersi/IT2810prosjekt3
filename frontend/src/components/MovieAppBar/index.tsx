import AppBar from "@mui/material/AppBar";
import Box from "@mui/material/Box";
import Toolbar from "@mui/material/Toolbar";
import IconButton from "@mui/material/IconButton";
import Typography from "@mui/material/Typography";
import MenuIcon from "@mui/icons-material/Menu";
import SearchBar from "../SearchBar";
import MovieFilter from "../MovieFilter";

export default function MovieAppBar() {
  return (
    <Box sx={{ flexGrow: 1 }}>
      <AppBar position="static">
        <Toolbar>
          <Typography
            variant="h6"
            noWrap
            component="div"
            sx={{ flexGrow: 1, display: { xs: "none", sm: "block" } }}
          >
            Movies
          </Typography>
          <MovieFilter />
          <SearchBar />
        </Toolbar>
      </AppBar>
    </Box>
  );
}
