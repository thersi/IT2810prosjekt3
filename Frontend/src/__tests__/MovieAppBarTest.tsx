import EnzymeToJson from "enzyme-to-json";
import { configure, render, shallow } from "enzyme";
import MovieAppBar from "../components/MovieAppBar";
import Adapter from "enzyme-adapter-react-16";

const component = (
  <MovieAppBar
    handleSearch={() => {
      return "";
    }}
    handleSort={() => {
      return 1;
    }}
    handleFilter={() => {
      return "";
    }}
  />
);
configure({ adapter: new Adapter() });
describe("Movie App Bar", () => {
  it("renders correctly", () => {
    render(component);
  });
  it("Should match snapshot", () => {
    const snapshotCheck = shallow(component);
    expect(EnzymeToJson(snapshotCheck)).toMatchSnapshot();
  });

  it("Should contain components", () => {
    const shallowComp = shallow(component);
    expect(shallowComp.find("MovieFilter").exists()).toBeTruthy();
    expect(shallowComp.find("SearchBar").exists()).toBeTruthy();
    expect(shallowComp.find("MovieDialog").exists()).toBeFalsy();
  });
});
