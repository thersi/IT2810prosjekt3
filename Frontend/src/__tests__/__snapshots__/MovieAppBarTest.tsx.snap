// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie App Bar Should match snapshot 1`] = `
<ForwardRef(Box)
  sx={
    Object {
      "flexGrow": 1,
    }
  }
>
  <ForwardRef(AppBar)
    position="static"
  >
    <ForwardRef(Toolbar)>
      <ForwardRef(Typography)
        component="div"
        noWrap={true}
        sx={
          Object {
            "display": Object {
              "sm": "block",
              "xs": "none",
            },
            "flexGrow": 1,
          }
        }
        variant="h6"
      >
        Movies
      </ForwardRef(Typography)>
      <MovieFilter
        aria-label="Movie filter to filter movies by ascending or descending order. Also filters by title og year. "
        aria-required="true"
        handleFilter={[Function]}
        handleSort={[Function]}
      />
      <SearchBar
        aria-label="Search bar to search through movie titles"
        aria-required="true"
        handleSearch={[Function]}
      />
    </ForwardRef(Toolbar)>
  </ForwardRef(AppBar)>
</ForwardRef(Box)>
`;
